src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +106,37 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    void checkoutIfNecessary(javax.jcr.Node node) throws RepositoryException {
+        try{
+            javax.jcr.Node versionableNode = findVersionableAncestor(node);
+            if (versionableNode != null) {
+                if (!versionableNode.isCheckedOut()) {
+                    versionableNode.checkout();
+                }
+            }
+        }
+        catch (Exception exception){
+            log.warn "Could not checkout node ${node.path}"
+        }
+
+    }
+
+    private javax.jcr.Node findVersionableAncestor(javax.jcr.Node node) throws RepositoryException {
```

Comment:
If a node is checked in, its connected non-versionable subtree become read-only. So if I need to modify any property of a node in this subtree, I would need to find versionable ancestor and perform checkout..
Moreover, if current node is versionable, this method is as good as just performing checkout on that node and not going to its parent